mysqlnd_azure.c

```diff
@@ -366,13 +406,19 @@ MYSQLND_METHOD(mysqlnd_azure_data, connect)(MYSQLND_CONN_DATA ** pconn,
             enum_func_status ret = FAIL;
             MYSQLND* redirect_conneHandle = mysqlnd_init(MYSQLND_CLIENT_KNOWS_RSET_COPY_DATA, conn->persistent); //init MYSQLND but only need only MYSQLND_CONN_DATA here
             if(!redirect_conneHandle) {
+              if (MYSQLND_AZURE_G(logLevel) > 0) {
+                AZURE_LOG("DEBUG", "Cannot connet with refirection information, redirect connection handshake failed");
+              }
                 DBG_ENTER("[redirect]: init redirect_conneHandle failed");
                 if(MYSQLND_AZURE_G(enableRedirect) == REDIRECT_ON) {
                     //REDIRECT_ON, abort the original connection and return error
                     conn->m->send_close(conn);
                     SET_CLIENT_ERROR(conn->error_info, MYSQLND_AZURE_ENFORCE_REDIRECT_ERROR_NO, UNKNOWN_SQLSTATE, "Connection aborted because init redirection failed.");
                     goto err;
                 } else {
+                  if (MYSQLND_AZURE_G(logLevel) > 1) {
+                    AZURE_LOG("INFO ", "Redirection connection failed, mysqlnd_azure.enableRedirect = PREFERRED, so fall back to classical connection.");
```

Comment:
Same here, the handshake is not started yet. So maybe we should state the error: " Init redirection object failed" instead of "Redirection connection failed"